src/queue.c

```diff
@@ -1,4 +1,9 @@
-#include <queue.h>
+#include <errno.h>
+#include <pthread.h>
+#include <sys/time.h>
```

Comment:
Pretty certain we need
`#include <stdint.h>`
here, as we use int64_t later. Stdint is not a standard include in many C libraries.